src/com/belsoft/project/CustomDictionary.java

```diff
@@ -0,0 +1,64 @@
+package com.belsoft.project;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomDictionary {
+
+    private final Map<String, String> names = new HashMap<>();
+    private final Map<String, String> ports = new HashMap<>();
+    private final Map<String, String> ips = new HashMap<>();
+
+    public void add(String key, String value) {
+        if (PortValidator.isValid(key)) {
```

Comment:
From what I see, if I am to add new possible key-value pairs, I have to add another 'else if' branch. This is valid Java code, but from an OOP perspective, I think this should be reconsidered.